### DISCUSSION TITLE
async support

### DISCUSSION THREAD

**Discussion by Matvii-Boichenko** [pallets-eco/flask-sqlalchemy#1273] [Q&A] [answered]
It would be nice to have instruction how to use SQLAlchemy async session with asyncpg with Flask-SQLAlchemy. I saw an [issue](https://github.com/pallets-eco/flask-sqlalchemy/issues/1031) opened in 2022 with similar request. It was closed because there were no need to rewrite library with async approach, however it was said that SQLAlchemy async session is already possible to implement with current library version overriding several methods. So, it would be very useful to get instruction on what exactly implementation is suggested.

**Comment by nioncode** [comment]
Did you find out how to override the engine to return an async engine? I only found how to override the arguments that are passed to `create_engine`, but we need to call `create_async_engine`.

**Comment by davidism** [comment, marked as answer]
You'll probably want to go with [Flask-SQLAlchemy-Lite](https://flask-sqlalchemy-lite.readthedocs.io/en/latest/) at this point. It is an official Pallets-Eco project just like this library. It tries to do as little as possible and as close to plain SQLAlchemy as possible, while still providing config and session management, including async sessions.
